adapters/consumable/consumable.go

```diff
@@ -0,0 +1,288 @@
+*/
+func (a *ConsumableAdapter) MakeBids(
+	internalRequest *openrtb.BidRequest,
+	externalRequest *adapters.RequestData,
+	response *adapters.ResponseData,
+) (*adapters.BidderResponse, []error) {
+
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	}
+
+	var serverResponse bidResponse // response from Consumable
+	if err := json.Unmarshal(response.Body, &serverResponse); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("error while decoding response, err: %s", err),
+		}}
+	}
+
+	bidderResponse := adapters.NewBidderResponse()
+	var errors []error
+
+	for impID, decision := range serverResponse.Decisions {
+		imp := getImp(impID, internalRequest.Imp)
+		if imp == nil {
+			errors = append(errors, &errortypes.BadServerResponse{
+				Message: fmt.Sprintf("ignoring bid id=%s, request doesn't contain any impression with id=%s", "TODO: bid.ID", impID),
+			})
+			fmt.Printf("%s", errors[0])
+			continue
+		}
+
+		if decision.Pricing != nil && decision.Pricing.ClearPrice != nil {
+
+			bid := openrtb.Bid{}
+			bid.ID = internalRequest.ID
+			bid.ImpID = impID
+			bid.Price = *decision.Pricing.ClearPrice
+			bid.AdM = retrieveAd(decision)
+			bid.W = imp.Banner.Format[0].W // TODO: Review to check if this is correct behaviour
+			bid.H = imp.Banner.Format[0].H
+			bid.CrID = strconv.FormatInt(decision.AdID, 10)
+			bid.Exp = 30 // TODO: Check this is intention of TTL
+
+			// not yet ported from prebid.js adapter
+			//bid.requestId = bidId;
+			//bid.currency = 'USD';
+			//bid.netRevenue = true;
+			//bid.referrer = utils.getTopWindowUrl();
+
+			bidderResponse.Bids = append(bidderResponse.Bids, &adapters.TypedBid{
+				Bid:     &bid,
+				BidType: getMediaTypeForImp(getImp(bid.ImpID, internalRequest.Imp)),
+			})
+		}
+	}
+	return bidderResponse, errors
+}
+
+func getImp(impId string, imps []openrtb.Imp) *openrtb.Imp {
+	for _, imp := range imps {
+		if imp.ID == impId {
+			return &imp
+		}
+	}
+	return nil
+}
+
+func extractExtensions(impression openrtb.Imp) (*adapters.ExtImpBidder, *openrtb_ext.ExtImpConsumable, []error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(impression.Ext, &bidderExt); err != nil {
+		fmt.Printf("err: %s", err) // TODO: DELETE
+		return nil, nil, []error{&errortypes.BadInput{
+			Message: err.Error(),
+		}}
+	}
+
+	var consumableExt openrtb_ext.ExtImpConsumable
+	if err := json.Unmarshal(bidderExt.Bidder, &consumableExt); err != nil {
+		fmt.Printf("err: %s", err) // TODO: DELETE
```

Comment:
And here

Reply:
Hmm.. I don't think I get a medal for this.

Sorry.. you shouldn't be left picking this up at review.  I'll run everything up this evening.  I just did a search for println() before ...